ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Mempool/Impl/Common.hs

```diff
@@ -82,6 +83,9 @@ data InternalState blk = IS {
       -- This should always be in-sync with the transactions in 'isTxs'.
     , isTxIds        :: !(Set (GenTxId blk))
 
+    , isTxKeys       :: !(LedgerTables (LedgerState blk) KeysMK)
```

Comment:
These two fields definitely need comments, which can probably state a pretty simple INVARIANT with respect to `isTxs`. I'd imagine it can be exact for `isTxKeys` and just a little hand-wavy for `isTxValues`.